### DISCUSSION TITLE
What is this HTTP server ?

### DISCUSSION THREAD

**Discussion by rnza0u** [prisma/prisma#8754] [Q&A] [answered]
My app is a NestJS app and it uses Prisma client.

An HTTP server is spawning on some random port after the prisma client connects to the database.

```
prisma:info Started http server on http://127.0.0.1:36917
```

It returns 404 when i send a request to `/`

I have absolutely no idea what this is. :( 

I searched the docs and found nothing about this server.

What is the purpose of this server ? Also, how could it be disabled in production ?

Thanks a lot.

**Comment by blevine** [comment]
I'd also be really interested in knowing the answer to this. Is Prisma using a local HTTP server as an intermediary between the client code and database?

**Reply by rnza0u** [reply]
See the answer below. Yes, it's the current way to communicate with the Rust low-level engine.

**Comment by ryands17** [comment, marked as answer]
The HTTP server is for `PrismaClient` to communicate with the Query Engine as explained [here](https://www.prisma.io/docs/concepts/components/prisma-engines/query-engine#the-query-engine-at-runtime).

To make this efficient without a server, you can enable [Node-API](https://www.prisma.io/docs/concepts/components/prisma-engines/query-engine#enable-the-node-api-n-api-preview) and that should reduce the overhead.

**Reply by rnza0u** [reply]
I had to check out the engine source code to realize what was going on because this documentation page **_does not_** talk about any HTTP server at all. 

It is really not clear about it.

Indeed, this seems like some big overhead.

**Reply by pantharshit00** [reply]
There is on going work for making Node API as default: https://github.com/prisma/prisma/issues/8731

When we started Prisma there were no good solution for Rust <-> Node.js bindings via Node API and using node-gyp wasn't a good solution. Now we using https://napi.rs. We intent to make Node API as default with 3.0.0

**Reply by blevine** [reply]
FWIW, switching to Node-API increased performance of my queries by about 2x.  Nice!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
